app/models/Comment.php

```diff
@@ -0,0 +1,136 @@
+<?php
+namespace app\models;
+use app\core\Database;
+use app\interfaces\EntityInterface;
+use InvalidArgumentException;
+
+class Comment implements EntityInterface
+{
+    protected Database $db;
+
+    public function __construct() {
+        $this->db = new Database();
+    }
+    /** API Create a new Post
+     * @param array $data
+     * @return void
+     */
+    public function create(array $data): void
+    {
+        $this->db->query('INSERT INTO Comments (post_id, name, message, user_id, created_at, status) VALUES (:post_id,:name, :message, :userId, NOW(), "pending")');
+        $this->db->bind('post_id', $data['postId']);
+        $this->db->bind(':name', $data['name']);
+        $this->db->bind(':message',$data['message']);
+        $this->db->bind(':userId', $data['userId']);
+        $this->db->execute();
+    }
+
+    /** Model delete a post from the DB
+     * @param $id
+     * @return void
+     */
+    public function delete($id): void
+    {
+        $sql = "DELETE FROM Comments WHERE id = :id";
+        $this->db->query($sql);
+        $this->db->bind(':id', $id);
+        $this->db->execute();
+    }
+
+    /** Model update a comment based on its id
+     * @param $id
+     * @param array $data
+     * @return void
+     */
+    public function update($id, array $data): void
+    {
+        // Prepare the SQL query
+        $this->db->query("UPDATE Comments SET name = :name, message = :message WHERE id = :commentId");
+
+        // Bind the parameters
+        $this->db->bind(':name', $data['name']);
+        $this->db->bind(':message', $data['message']);
+        $this->db->bind(':commentId', $data['commentId']);
+
+        $this->db->execute();
+    }
+
+    /** Model get a comment based on its ID
+     * @param $id
+     * @return mixed
+     */
+    public function getCommentByID($id)
```

Comment:
this function is very similar to `getPostsById` in the Post model.  Could we leverage the Entity interface to have this generic select mechanism occur in a parent class?  Same thing for other standard commands such as `delete`